### PR TITLE
Tostring

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
Bind __tostring instead of toString for Lua. This way stringification works automatically.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
I don't think we should remove the existing `toString` binding, otherwise we will break existing code/configurations.

**Comment by cmouse** [comment]
possibly, yes. I can restore them if the tests pass.

**Comment by cmouse** [comment]
Travis CI failure is due to missing python3

File: pdns/dnsdist-lua-rules.cc [hunk 1]
```diff
@@ -479,6 +479,10 @@ void setupLuaRules()
       return std::dynamic_pointer_cast<DNSRule>(tisr);
     });
 
+  g_lua.registerFunction<void(std::shared_ptr<TimedIPSetRule>::*)()>("__toString", [](std::shared_ptr<TimedIPSetRule> tisr) {
```
Comment (zeha) [inline pdns/dnsdist-lua-rules.cc:482]: shouldn't this also be `__tostring` (lowercase) then?

Comment (cmouse) [reply pdns/dnsdist-lua-rules.cc]: Thanks.

**Comment by cmouse** [comment]
> 
> 
> I don't think we should remove the existing `toString` binding, otherwise we will break existing code/configurations.

Fixed.

**Review by rgacogne — APPROVED** [review]

**Comment by Habbie** [comment]
Nice! Can you resolve the conflicts so that we can merge this?

**Comment by pieterlexis** [comment]
could you rebase again? (sorry)

**Comment by pieterlexis** [comment]
could you rebase again? (sorry)

**Comment by cmouse** [comment]
done.

**Review by rgacogne — APPROVED** [review]

**Pull request merged**